### DISCUSSION TITLE
How to link a user account from different providers

### DISCUSSION THREAD

**Discussion by adrianfdez469** [nextauthjs/next-auth#1390] [Help] [answered]
**Your question**
What can I do to link a user account by email from different providers ( trusted providers that has email verification to signup)
I read the [why are accounts not linked automatically](https://next-auth.js.org/faq#when-i-sign-in-with-another-account-with-the-same-email-address-why-are-accounts-not-linked-automatically) from NextAuth.js Frequently Asked Questions, and I want to do it any ways with some serious/trusted providers, but don't know how. I check the docs but don't see anything about it. Is there any code example/guide/article that explain how to achieve this?

**What are you trying to do**
I'm developing a web app, and I want to link users accounts by email, regardless the provider they choose. ( The providers are all from trusted sources )

**Reproduction**
I'm using `"next-auth": "^3.3.0-canary.11"`, PostgresSQL, Prisma adapter and `session: { jwt: true }`

By the way, awesome work with the library. 👍

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by balazsorban44** [comment, marked as answer]
https://github.com/nextauthjs/next-auth/pull/1002#issuecomment-765590415 So I guess this would be relevant for you.

PS.: ⚠ Please upgrade to the latest `next-auth` version ⚠ (which is `3.6.0` as of this writing). A Security issue has been patched since your version, and so it is strongly advised to upgrade!

See: [Insecure email token verification in Prisma adapter
](https://github.com/nextauthjs/next-auth/security/advisories/GHSA-pg53-56cg-4m8q)
